### PR TITLE
Add documentation for container checkpointing feature (KEP 2008)

### PR DESCRIPTION
**Pull request by adrianreber**
Pull request for KEP 2008:

* https://github.com/kubernetes/enhancements/pull/1990
* https://github.com/kubernetes/enhancements/issues/2008
* https://github.com/kubernetes/kubernetes/pull/104907
* https://github.com/kubernetes/website/pull/31753 (previous PR for 1.24 (not merged))

CC: @sftim (you reviewed my previous 1.24 attempt of this documentation PR)

### CODE REVIEW

**Comment by adrianreber** [comment]
CC: @mikebrow

**Review by sftim — COMMENTED** [review]
Based on:
>  TODO: Add more information about return codes once CRI implementation have checkpoint/restore.

/retitle [WIP] Add documentation for container checkpointing feature (KEP 2008)

**Comment by adrianreber** [comment]
> Based on:
> 
> > TODO: Add more information about return codes once CRI implementation have checkpoint/restore.
> 
> /retitle [WIP] Add documentation for container checkpointing feature (KEP 2008)

We will not be able to fix this TODO before the release. Without the release the new CRI API is not available and cannot be implemented by the CRI implementation. This TODO can only be fixed after the release.

**Review by chrisnegus — COMMENTED** [review]
This reads really well. I was hoping you could add a bit more in your return codes comment.

**Comment by reylejano** [comment]
@adrianreber , when this PR is ready for review, please remove [WIP] from the title
/milestone 1.25
/assign @sethmccombs 
/cc @kcmartin

File: content/en/docs/reference/node/kubelet-checkpoint-api.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+500: Internal Server Error (if the CRI implementation encounter an error during checkpointing (see error message for further details))
+
+500: Internal Server Error (if the CRI implementation does not implement the checkpoint CRI API (see error message for further details))
+
+{{< comment >}}
+TODO: Add more information about return codes once CRI implementation have checkpoint/restore.
+      This TODO cannot be fixed before the release, because the CRI implementation need
+      the Kubernetes changes to be merged to implement the new CheckpointContainer CRI API
+      call. We need to wait after the 1.25 release to fix this.
+{{< /comment >}}
```
Comment (reylejano) [inline content/en/docs/reference/node/kubelet-checkpoint-api.md:96]: I suggest removing this section and create a GitHub issue to track this
```suggestion
```

**Comment by reylejano** [comment]
Hi @adrianreber , the 1.25 Docs complete deadline passed. 
Please address comments and feedback on this PR as soon as possible

**Comment by reylejano** [comment]
I can address my comment in a follow-up PR
/approve

**Comment by adrianreber** [comment]
Sorry. I was away for some time and not able to react in a timely manner. I can do any fixup needed if you let me know what you would like to have changed.

**Pull request merged**